crypto/src/ed25519.rs

```diff
@@ -197,6 +291,38 @@ impl KeyPair for Ed25519KeyPair {
             secret: Ed25519PrivateKey(kp.secret),
         }
     }
+
+    fn deterministic_generate(seed: &[u8], id: &[u8], domain: &[u8]) -> Result<Self, signature::Error> {
```

Comment:
I like that you're proactive about this and not leaving the user to figure out KDF on their own.

But here we end up essentially copying the implementation from the BLS variant, which hurts maintenability.

Here's one approach that leads to one version of this code:
- a generic variant of the KDF, that's just parametrized in each implementation,
- or simply an `ikm -> KeyPair` map on the trait, that's then used at the end of a [default implementation](https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations) leveraging the KDF.